models/classes/AuthoringAsTool/AuthoringAsToolLtiConfigProvider.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes\AuthoringAsTool;
+
+use oat\oatbox\session\SessionService;
+use oat\tao\model\AuthoringAsTool\AuthoringAsToolConfigProviderInterface;
+use oat\taoLti\models\classes\LtiLaunchData;
+use oat\taoLti\models\classes\TaoLtiSession;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class AuthoringAsToolLtiConfigProvider implements AuthoringAsToolConfigProviderInterface
+{
+    private AuthoringAsToolConfigProviderInterface $configFallback;
+
+    private SessionService $session;
+    private LoggerInterface $logger;
+
+    public function __construct(
+        AuthoringAsToolConfigProviderInterface $configFallback,
```

Comment:
You are providing AuthoringAsToolConfigProviderInterface implementation as a dependency property in a service that is implementing this same interface?

Reply:
Why don't you overload AuthoringAsToolConfigProviderInterface implementation in ServiceProvider with new service in taoLti that extends from tao-core service that implements AuthoringAsToolConfigProviderInterface?